std/range/primitives.d

```diff
@@ -99,6 +99,29 @@ $(BOOKTABLE ,
     ))
 )
 
+Templates that test for various output _range capabilities:
+
+$(BOOKTABLE ,
+    $(TR $(TD $(D $(LREF hasStart)))
+        $(TD Tests if a given output _range has the $(D start) function.
+    ))
+    $(TR $(TD $(D $(LREF hasFinish)))
+        $(TD Tests if a given output _range has the $(D finish) function.
+    ))
+)
+
+Convenience functions for manipulating output ranges:
+
+$(BOOKTABLE ,
+    $(TR $(TD $(D finish))
+        $(TD Tells output range that no more input is coming,
+        optionally returns the digest of the data.
```

Comment:
I meant 'digest' in a generic sense, meaning anything the output range author thinks is appropriate.